rfcs/90-parse-pairs-from-string.md

```diff
@@ -0,0 +1,106 @@
+---
+author: xxchan <xxchan22f@gmail.com>
+status: draft
+updated_at: 2021-06-01
+deprecates:
+  - 13-remove-config-string.md
+---
+
+# GSP-90: Parse Pairs from String
```

Comment:
This GSP intends to replace config string, it's not a general parser that parses the string into pairs.

Reply:
Why not? In my imagination, we support general parsing here and then config string follows.

Reply:
Let's build features that users need. For now, I don't think the user needs a pair parser except in service/storage initialization. Perhaps we can implement it when we find that there is a real need for it.

Reply:
I agree, but I thought if we provide API like `FromMap`, isn't it already a general pair parser? Users can use this API anywhere, although most often in initialization.

Reply:
It is indeed easy to misunderstand.

How about adding APIs like `NewStoragerFromString` which parse the string into pairs and call `NewStorager`? And we will not expose the API for parsing pairs until we figure this out.

Reply:
Then this API is effectively the old config string. We just polished the implementation.

Reply:
> Then this API is effectively the old init string. We just polished the implementation.

That's true.

Reply:
But the format could be redesigned.

Reply:
> But the format could be redesigned.

Oh actually this is what I wanted to consider later and that's why config string is not discussed in this proposal 🤔